core/core/src/sf_core.rs

```diff
@@ -128,12 +128,19 @@ enum MapCacheEntryError {
 struct MapCacheEntry {
     pub map: String,
     pub content_hash: String,
+    /// This is for the purposes of stacktraces in JsInterpreter
+    pub file_name: String
 }
 impl MapCacheEntry {
-    pub fn from_data(data: Vec<u8>) -> Result<Self, MapCacheEntryError> {
+    // TODO: name should be taken from the manifest
+    pub fn new(data: Vec<u8>, file_name: String) -> Result<Self, MapCacheEntryError> {
```

Comment:
unrelated to pr: `from_data` vs `new` what is best practice? I kinda fight it right now.

Reply:
I see, one vs. more parameters sounds as good devider.